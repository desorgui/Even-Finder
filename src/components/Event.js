import React from 'react';
import PropTypes from 'prop-types';
import { AiOutlineCalendar } from 'react-icons/ai';
import { BsCash } from 'react-icons/bs';

const Event = ({
  id, name, image, classification, localDate, priceRanges,
}) => (
  <div className="singleEvent" id={id} key={id}>
    <div className="event-inner">
      <div className="event-event-style-3" data-event-id="143">
        <div className="event-image">
          <img src={`${image}`} alt={`${name}`} className="event-img" />
        </div>
        <div className="event-content">
          <div className="event-title">
            {name}
          </div>
          <div className="event-details">
            <div className="event-category">
              {classification}
            </div>
            <div className="event-date event-start-date">
              <span>
                <AiOutlineCalendar />
                {' '}
                {localDate}
              </span>
            </div>
          </div>
          <div className="event-details">
            <div className="event-status">
              {priceRanges !== 'undefined' ? (
                <span>
                  <BsCash />
                  {' '}
                  $
                  {priceRanges[0].min}
                  {' '}
                  - $
                  {priceRanges[0].max}
                </span>
              ) : (
                <span>
                  <BsCash />
                  {' '}
                  Free
                </span>
              )}
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
);

Event.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  classification: PropTypes.string.isRequired,
  localDate: PropTypes.string.isRequired,
  priceRanges: PropTypes.string.isRequired,
};

export default Event;
