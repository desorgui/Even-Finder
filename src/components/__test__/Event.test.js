import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import store from '../../redux/configureStore';
import Event from '../Event';

function renderWithProviders(
  ui,
  {
    storetest = store,
    ...renderOptions
  } = {},
) {
    function Wrapper({ children }) { // eslint-disable-line
    return <Provider store={storetest}>{children}</Provider>;
  }
  return { storetest, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

const event = {
  id: 'Z7r9jZ1AdFYe3',
  name: 'Los Angeles Dodgers vs. San Diego Padres',
  localDate: '2022-08-05',
  localTime: '19:10:00',
  image: 'https://s1.ticketm.net/dam/a/2f4/5f8bca36-3127-4f87-b573-3dd6485642f4_1344231_RETINA_PORTRAIT_3_2.jpg',
  classification: 'Sports',
  priceRanges: 'undefined',
};

it('Event renders information correctly', () => {
  renderWithProviders(<Event {...event} />); // eslint-disable-line
  expect(screen.getByText('Los Angeles Dodgers vs. San Diego Padres')).toBeInTheDocument();
});

test(' Test the snapshot', () => {
  expect(screen.debug()).toMatchSnapshot();
});
