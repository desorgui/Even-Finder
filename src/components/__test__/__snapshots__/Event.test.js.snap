// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Test the snapshot 1`] = `undefined`;
